src/svc_vc.c

```diff
@@ -526,12 +526,20 @@ svc_vc_destroy_task(struct work_pool_entry *wpe)
 	struct rpc_dplx_rec *rec =
 			opr_containerof(wpe, struct rpc_dplx_rec, ioq.ioq_wpe);
 
+	__warnx(TIRPC_DEBUG_FLAG_REFCNT,
+		"%s() %p fd %d xp_refs %" PRIu32,
+		__func__, rec, rec->xprt.xp_fd, rec->xprt.xp_refs);
+
 	if (rec->xprt.xp_refs) {
 		/* instead of nanosleep */
 		work_pool_submit(&svc_work_pool, &(rec->ioq.ioq_wpe));
 		return;
 	}
 
+	if ((rec->xprt.xp_flags & SVC_XPRT_FLAG_CLOSE)
+	    && rec->xprt.xp_fd != RPC_ANYFD)
+		(void)close(rec->xprt.xp_fd);
```

Comment:
Ditto here (and both places should probably use the same constant, either -1 or RPC_ANYFD)